keyring/backend.py

```diff
@@ -144,14 +144,11 @@ def get_credential(
     def set_properties_from_env(self) -> None:
         """For all KEYRING_PROPERTY_* env var, set that property."""
 
-        def parse(item: typing.Tuple[str, str]):
-            key, value = item
-            pre, sep, name = key.partition('KEYRING_PROPERTY_')
-            return sep and (name.lower(), value)
-
-        props: filter[typing.Tuple[str, str]] = filter(None, map(parse, os.environ.items()))
-        for name, value in props:
-            setattr(self, name, value)
+        for key, value in os.environ.items():
+            _, sep, name = key.partition('KEYRING_PROPERTY_')
+            if not sep:
+                continue
+            setattr(self, name.lower(), value)
```

Comment:
I'd prefer to use a functional paradigm. I do recognize how the imperative approach is simpler, especially when type annotations are present. Still, it's important for me when adding typing hints not to allow the addition of hints to change the semantic meaning of the implementation, so I've backed out this commit.

Reply:
fine by me, but what do you mean by "backed out this commit", nothing is backed out, its still here.

Do you mean i should remove it?

Reply:
Ah sorry see it now, you reverted it, i would find it nicer if i simply remove the commit if its fine by you.

Reply:
Yes, feel free to rewrite the history without those changes.